tools/shared.py

```diff
@@ -2635,14 +2632,37 @@ def metadce(js_file, wasm_file, minify_whitespace, debug_info):
     extra_info = {'unused': unused}
     return Building.acorn_optimizer(js_file, passes, extra_info=json.dumps(extra_info))
 
+  @staticmethod
+  def asyncify_lazy_load_code(wasm_binary_target, options, debug):
+    # create the lazy-loaded wasm. remove the memory segments from it, as memory
+    # segments have already been applied by the initial wasm, and apply the knowledge
+    # that it will only rewind, after which optimizations can remove some code
+    cmd = Building.get_wasm_opt_command(debug=debug)
+    cmd += [wasm_binary_target, '-o', wasm_binary_target + '.lazy.wasm']
+    cmd += ['--remove-memory']
```

Comment:
Why this?

Reply:
The memory? See the comment a few lines up, `remove the memory segments from it, as memory segments have already been applied by the initial wasm`